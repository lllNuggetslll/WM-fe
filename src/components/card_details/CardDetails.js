import React from "react";
import { string, number, shape } from "prop-types";
import styled from "styled-components";
import { StarRating } from "./../star_rating";

const CardInfoWrapper = styled.div`
  text-align: left;
  margin-left: 20px;
  display: flex;
  flex-direction: column;
  height: 64px;
  justify-content: space-between;

  @media only screen and (max-width: 750px) {
    margin-left: 0;
  }
`;

const Name = styled.div`
  font-weight: bold;
  opacity: 0.6;
  white-space: nowrap;
`;

const CardDetails = ({ listing }) => {
  return (
    <CardInfoWrapper>
      <div style={{ opacity: 0.4, display: "flex" }}>
        {listing.city} <hr style={{ margin: "0 5px" }} />{" "}
        {Math.round(listing.distance)}
        mi
      </div>
      <Name>{listing.name}</Name>
      <StarRating rating={listing.rating} />
    </CardInfoWrapper>
  );
};

CardDetails.propTypes = {
  listing: shape({
    city: string.isRequired,
    distance: number.isRequired,
    name: string.isRequired,
    rating: number.isRequired
  }).isRequired
};

export default CardDetails;
