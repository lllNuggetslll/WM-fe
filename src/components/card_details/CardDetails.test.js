import React from "react";
import { shallow } from "enzyme";
import CardDetails from "./CardDetails";

describe("CardDetails", () => {
  it("should render", () => {
    const listing = {
      city: "city",
      distance: 1,
      name: "name",
      rating: 1
    };
    const wrapper = shallow(<CardDetails {...{ listing }} />);

    expect(wrapper).toMatchSnapshot();
  });
});
