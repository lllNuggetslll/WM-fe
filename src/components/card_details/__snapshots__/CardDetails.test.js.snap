// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDetails should render 1`] = `
<styled.div>
  <div
    style={
      Object {
        "display": "flex",
        "opacity": 0.4,
      }
    }
  >
    city
     
    <hr
      style={
        Object {
          "margin": "0 5px",
        }
      }
    />
     
    1
    mi
  </div>
  <styled.div>
    name
  </styled.div>
  <StarRating
    rating={1}
  />
</styled.div>
`;
